src/distilabel/tasks/text_generation/evol_complexity.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Literal, Optional
+
+from distilabel.logger import get_logger
+from distilabel.tasks.base import get_template
+from distilabel.tasks.prompt import Prompt
+from distilabel.tasks.text_generation.evol_instruct import EvolInstructTask
+
+logger = get_logger()
+
+
+# inherits from evol-instruct with limited evolution methods
+_EVOL_COMPLEXITY_TEMPLATE = get_template("evol-instruct.jinja2")
+
+
+EvolutionMethod = Literal["constraints", "deepen", "concretizing", "reasoning"]
+
+
+@dataclass
+class EvolComplexityGeneratorTask(EvolInstructTask):
```

Comment:
I would change the name of this class to just simply `EvolComplexityTask` or if we want to keep the reference of `generation` then I would align the name with the `TextGenerationTask` and call it `EvolComplexityGenerationTask`.